### PR TITLE
docs: gatekeeper page

### LINKED ISSUE #1382: Add documentation of gatekeepers
# Motivation

I find myself often sharing a link to our gatekeeper contracts:
https://github.com/privacy-scaling-explorations/maci/tree/dev/contracts/contracts/gatekeepers

But there's no README on this directory or page in the [MACI documentation](https://maci.pse.dev/docs/introduction) that explains how gatekeepers work

# Details

Create a new documentation page that explains gatekeepers, how they work, how they can be configured, and how someone might build their own custom gatekeeper contract to use MACI in a new way. More broadly it might be worth having this page explain how identity/authentication/access control happens in MACI, how MACI doesn't aim to solve this problem directly but relies on other solutions (e.g. EAS, Hats Protocol, Gitcoin Passport, whatever solution) to manage the identity & how the MACI gatekeeper contract then uses that identity system or authentication process to register users as eligible to vote.

### PR DESCRIPTION
**Pull request by samajammin**
# Description

Add gatekeeper docs page

## Related issue(s)

Fixes #1382

## Confirmation

- [x] I have read and understand MACI's [contributor guidelines](https://maci.pse.dev/docs/contributing) and [code of conduct](https://maci.pse.dev/docs/contributing/code-of-conduct).
- [x] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
- [x] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

### CODE REVIEW

File: website/versioned_docs/version-v1.3_alpha/contracts/contracts.md [hunk 1]
```diff
@@ -329,22 +329,9 @@ This contract can be used by the `SignUpGateKeeper` to determine whether a user
 
 ## SignUpGatekeeper
 
-MACI requires a signup gatekeeper to ensure that only designed users register. It is up to MACI's deployer how they wish to allow sign-ups, therefore they can implement their own GateKeeper. The repository comes with different options:
+MACI requires a gatekeeper cont ract to ensure that only eligible users can register.
```
Comment (ctrlc03) [inline website/versioned_docs/version-v1.3_alpha/contracts/contracts.md:332]: seems like there's a space in the word "contract"

File: website/versioned_docs/version-v1.3_alpha/contracts/gatekeepers.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+title: MACI Gatekeepers
+description: MACI uses a "gatekeeper" contract, which is used to determine the eligibility criteria of voters who can participate in MACI polls
+sidebar_label: Gatekeepers
+---
+
+# MACI Gatekeepers
+
+MACI is composed of multiple smart contracts, which together with the zk-SNARK circuits, can be used to carry out on-chain voting. Start with an [overview of our smart contracts here](/docs/contracts).
+
+MACI uses a "gatekeeper" contract, which is used to configure and enforce the eligibility criteria of voters who can participate in MACI polls. In other words, it is a way to whitelist signups to the system. Operators of MACI can use the gatekeeper contract to configure signup eligibility and to protect agaisnt sybil attacks in their polls.
```
Comment (ctrlc03) [inline website/versioned_docs/version-v1.3_alpha/contracts/gatekeepers.md:11]: typo: "agaisnt"

File: website/versioned_docs/version-v1.3_alpha/contracts/gatekeepers.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+Developers are also welcome to implement a custom gatekeeper contract as long as it conforms to the [`SignUpGatekeeper`](https://github.com/privacy-scaling-explorations/maci/blob/dev/contracts/contracts/gatekeepers/SignUpGatekeeper.sol) - an abstract contract to inherit from with two function signatures:
+
+```ts
+abstract contract SignUpGatekeeper {
+    function setMaciInstance(MACI _maci) public virtual {}
+    function register(address _user, bytes memory _data) public virtual {}
+}
+```
+
+Whenever a user signs, up (via the `MACI.signUp` function), this triggers the MACI contract to call the `SignUpGatekeeper.register` function.
```
Comment (ctrlc03) [inline website/versioned_docs/version-v1.3_alpha/contracts/gatekeepers.md:30]: no comma between "signs" and "up"

Comment (ctrlc03) [inline website/versioned_docs/version-v1.3_alpha/contracts/gatekeepers.md:30]: it's important to add that the `SigupGatekeeper.register` function should either return successfully or revert on error - it should not be returning true or false

**Review by ctrlc03 — CHANGES_REQUESTED** [review]
Thanks! left some comments

**Comment by samajammin** [comment]
@ctrlc03 if you or someone else has bandwidth to make changes & merge, I'd be grateful 🙏

**Comment by ctrlc03** [comment]
> @ctrlc03 if you or someone else has bandwidth to make changes & merge, I'd be grateful 🙏

changes made, will need a review please

**Review by 0xmad — APPROVED** [review]
@samajammin @ctrlc03 thanks!

File: website/versioned_docs/version-v2.0_alpha/developers-references/smart-contracts/Gatekeepers.md [hunk 4]
```diff
@@ -5,15 +5,22 @@ sidebar_label: Gatekeepers
...
+- `SignUpTokenGatekeeper` - This restricts signups only to holders of a specific ERC721 token.
+- `EASGatekeeper` - This restricts signups only to users who have received an attestation of a specific schema from a trusted attester.
+- `HatsGatekeeper` - This allows gatekeeping signups to only users who have a specific [Hat](https://www.hatsprotocol.xyz/).
+- `GitcoinPassportGatekeeper` - This allows gatekeeping signups to only users who have a specific [Gitcoin Passport](https://passport.gitcoin.co/) score.
 
-An abstract contract to inherit from is also provided, with two function signatures as shown below:
+You can find all of our latest gatekeeper contracts here:
+https://github.com/privacy-scaling-explorations/maci/tree/dev/contracts/contracts/gatekeepers
+
+Developers are also welcome to implement a custom gatekeeper contracts as long as it conforms to the `SignUpGatekeeper` - an abstract contract to inherit from with two function signatures:
```
Comment (kittybest) [inline website/versioned_docs/version-v2.0_alpha/developers-references/smart-contracts/Gatekeepers.md:23]: nit: contracts --> contract

**Review by ctrlc03 — APPROVED** [review]

**Pull request merged**